nxdrive/engine/engine.py

```diff
@@ -435,6 +450,7 @@ def _direct_transfer(
         /,
         *,
         duplicate_behavior: str = "create",
+        new_folder: Optional[str] = None,
```

Comment:
Nitpick: move the line after `last_local_selected_location` to keep those sorted.